mypy/constraints.py

```diff
@@ -79,15 +79,19 @@ def __repr__(self) -> str:
         op_str = "<:"
         if self.op == SUPERTYPE_OF:
             op_str = ":>"
-        return f"{self.type_var} {op_str} {self.target}"
+        return f"{self.origin_type_var} {op_str} {self.target}"
```

Comment:
I'm not too sure why this was this way before and if this is entirely bad behavior. I added this (and updates to hashing and equality) because paramspec types have extra state that is important: their prefix (which `Concatenate` adds)